File: python-support/pil/db/Result.py

class Result(object):
    def __init__(self, rs):
        self.values = rs
	
    def getInt(self, index):
		return int(self.values[index])

    def getString(self, index):
		return str(values[index])
	}
	
    def getDateTime(self, index):
        return datetime.strptime('YYYY-MM-DD HH:MM:SS.mmmmm', str(values[index]))
	
    def __str__(self):
		return str(values)

File: java-support/src/pil/db/Result.java
package pil.db;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.text.ParseException;
import java.util.ArrayList;
import java.util.Date;

import pil.json.JsonObject;

public class Result {
	ArrayList<String> values = new ArrayList<String>();

	public Result(ResultSet rs) {
		try {
			int columns = rs.getMetaData().getColumnCount();
			for(int i = 0; i < columns; i++) {
				values.add(rs.getString(i+1));
			}
		} catch (SQLException e) {
			throw new RuntimeException(e);
		}
	}
	
	public int getInt(int index) {
		return Integer.valueOf(values.get(index));
	}

	public String getString(int index) {
		return values.get(index);
	}
	
	public Date getDateTime(int index) {
		try {
			return JsonObject.dateFormat.parse(values.get(index));
		} catch (ParseException e) {
			return null;
		}
	}
	
	public String toString() {
		return values.toString();
	}

}
